train.py

```diff
@@ -294,16 +303,24 @@ def main():
     losses, train_top1, prec_at_k, _, neg_dists, pos_dists = loss.LOSS_CHOICES[args.loss](
         dists, pids, args.margin, batch_precision_at_k=args.batch_k-1)
 
-    # Count the number of active entries, and compute the total batch loss.
-    num_active = tf.reduce_sum(tf.cast(tf.greater(losses, 1e-5), tf.float32))
-    loss_mean = tf.reduce_mean(losses)
+    # Count how many entries in the batch are (possibly approximately) non-zero.
+    if args.loss_ignore_zero is True:
+        nnz = tf.count_nonzero(losses, dtype=tf.float32)
+    else:
+        nnz = tf.reduce_sum(tf.to_float(tf.greater(losses, args.loss_ignore_zero or 1e-5)))
```

Comment:
Is the point of this supposed to be just for logging?

Reply:
No, actually. It's type-magic and can be a little obscure, hence why I still need to write documentation in the README :)

The `else` case happens when `loss_ignore_zero` is given an additional float argument, so one can call it as `--loss_ignore_zero 1e-3` for example, in order to consider anything below 1e-3 to be counted as zero.

Reply:
Just trying to get my head around this. Why would making loss' that aren't zero, zero help anything?

Reply:
Read our paper, we explain them in there :) But really it's not a good time investment to play with that parameter.